### PR TITLE
return public_key for clients in all cases

### PR DESCRIPTION
**Pull request by sdelano**
To be consistent with the open-source API and tools that are depending
on it, we should always return public keys with client objects instead
of certificates. The following is a situation that we've recently
encountered:
- consumer of API switches on the response having json_class field
- response inflates to a Chef::ApiClient if json_class is set
- if inflated, consumer expects the public_key field to be set
- if not inflated (no json_class), consumer expects certificate
  field to be set

This commit ensures that we always set both json_class and
public_key, the combination that the Ruby chef-client code expects.

### CODE REVIEW

File: src/chef_client.erl [hunk 1]
```diff
@@ -229,21 +229,13 @@ assemble_client_ejson(Client, OrgName) ->
...
-                    {[{<<"public_key">>, PublicKey} | Values]};
-                1 ->
-                    {[{<<"certificate">>, PublicKey} | Values]}
-            end
-    end.
+    {[{<<"name">>, value_or_default(Name, <<"">>)},
+      {<<"clientname">>, value_or_default(Name, <<"">>)},
+      {<<"validator">>, Validator =:= true},
+      {<<"orgname">>, OrgName},
+      {<<"public_key">>, chef_object_base:extract_public_key(PublicKey)},
```
Comment (jmink) [inline src/chef_client.erl]: I think this can include an undefined public key.  Is that desired?

Comment (sdelano) [inline src/chef_client.erl]: Nice catch. That functionality is also in the `osc_` part. Unfortunately none of our unit / functional tests caught this, so I'll rectify that as well.

File: src/chef_client.erl [hunk 2]
```diff
@@ -232,17 +232,14 @@ oc_assemble_client_ejson(#chef_client{name = Name, validator = Validator,
     Values = [{<<"name">>, value_or_default(Name, <<"">>)},
               {<<"clientname">>, value_or_default(Name, <<"">>)},
               {<<"validator">>, Validator =:= true},
-              {<<"orgname">>, OrgName}],
+              {<<"orgname">>, OrgName},
+              {<<"json_class">>, <<"Chef::ApiClient">>},
+              {<<"chef_type">>, <<"client">>}],
     case PublicKey of
         undefined ->
             {Values};
```
Comment (marcparadise) [inline src/chef_client.erl:240]: What will clients do when this is missing as opposed to null?

Comment (sdelano) [inline src/chef_client.erl:240]: I just added this back in for compatibility with open source and the previous code. I do not know.

Comment (sdelano) [inline src/chef_client.erl:240]: We're also severely lacking integration tests around the client objects in pedant.

Comment (markan) [inline src/chef_client.erl:240]: Does any of the logic around DEFAULT_FIELD_VALUES and OC_DEFAULT_FIELD_VALUES in *_parse_binary_json need updating for consistency?

**Comment by sdelano** [comment]
Closing because this will be merged in with #77

**Pull request closed without merging**